### PR TITLE
Renaming existing staking token to base unit

### PR DESCRIPTION
**Pull request by tjanez**
Introduce `TokenSymbol` and `TokenValue` staking parameters which represent the token's ticker symbol and token's value in base units.

TODO:
- [x] Add sanity checks for `TokenSymbol` and `TokenValue`.

### CODE REVIEW

File: go/consensus/tendermint/staking/staking.go [hunk 1]
```diff
@@ -38,6 +38,24 @@ type tendermintBackend struct {
...
+	return genesis.Staking.TokenSymbol, nil
+}
+
+func (tb *tendermintBackend) TokenValue(ctx context.Context) (*quantity.Quantity, error) {
+	genesis, err := tb.service.GetGenesisDocument(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &genesis.Staking.TokenValue, nil
```
Comment (kostko) [inline go/consensus/tendermint/staking/staking.go:56]:
```suggestion
	return genesis.Staking.TokenValue.Clone(), nil
```

File: go/staking/api/api.go [hunk 2]
```diff
@@ -642,11 +648,14 @@ type DebondingDelegation struct {
 	DebondEndTime epochtime.EpochTime `json:"debond_end"`
 }
 
-// Genesis is the initial ledger balances at genesis for use in the genesis
-// block and test cases.
+// Genesis is the initial staking state for use in the genesis block.
 type Genesis struct {
 	Parameters ConsensusParameters `json:"params"`
 
+	// TokenSymbol is the token's ticker symbol.
```
Comment (kostko) [inline go/staking/api/api.go:655]: This grouping looks strange, either move these to their own group or add comments everywhere.

File: go/consensus/tendermint/staking/staking.go (hunk 1 above)
Comment (tjanez) [reply go/consensus/tendermint/staking/staking.go]: Fixed.

File: go/staking/api/api.go (hunk 2 above)
Comment (tjanez) [reply go/staking/api/api.go:656]: I've added comments for all the fields.

File: .changelog/3061.feature.2.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+go/oasis-node/cmd/genesis: Allow setting staking token symbol in init
+
+Allow setting staking token's ticker symbol in `oasis-node genesis init` CLI
```
Comment (kostko) [inline .changelog/3061.feature.2.md:3]: But not the exponent?

File: docs/consensus/staking.md [hunk 4]
```diff
@@ -13,6 +13,25 @@ the [consensus service API documentation].
 [consensus service API documentation]: https://pkg.go.dev/github.com/oasisprotocol/oasis-core/go/staking/api?tab=doc
 <!-- markdownlint-enable line-length -->
 
+## Tokens and base units
```
Comment (kostko) [inline docs/consensus/staking.md:16]:
```suggestion
## Tokens and Base Units
```

File: go/oasis-test-runner/oasis/oasis.go [hunk 5]
```diff
@@ -802,6 +802,7 @@ func (net *Network) makeGenesis() error {
 		"--" + genesis.CfgRegistryDebugAllowTestRuntimes, "true",
 		"--scheduler.max_validators_per_entity", strconv.Itoa(len(net.Validators())),
 		"--" + genesis.CfgConsensusGasCostsTxByte, strconv.FormatUint(net.cfg.ConsensusGasCostsTxByte, 10),
+		"--staking.token_symbol", "TEST",
```
Comment (kostko) [inline go/oasis-test-runner/oasis/oasis.go:805]: Use config option name constant?

File: go/staking/api/sanity_check.go [hunk 6]
```diff
@@ -228,6 +229,22 @@ func (g *Genesis) SanityCheck(now epochtime.EpochTime) error { // nolint: gocycl
...
 	}
 
+	tokenSymbolLength := len(g.TokenSymbol)
+	if tokenSymbolLength == 0 {
+		return fmt.Errorf("staking: sanity check failed: token symbol is empty")
+	}
+	if tokenSymbolLength > TokenSymbolMaxLength {
+		return fmt.Errorf("staking: sanity check failed: token symbol exceeds maximum length")
+	}
+	match := regexp.MustCompile("^[A-Z]+$").FindString(g.TokenSymbol)
```
Comment (kostko) [inline go/staking/api/sanity_check.go:239]: This regexp should probably be an exported variable at the api level (together with the max length/max value.

File: .changelog/3061.feature.2.md (hunk 3 above)
Comment (tjanez) [reply .changelog/3061.feature.2.md]: Good catch. I've added the ability to set the token value's exponent.

File: docs/consensus/staking.md (hunk 4 above)
Comment (tjanez) [reply docs/consensus/staking.md]: Fixed.

File: go/oasis-test-runner/oasis/oasis.go (hunk 5 above)
Comment (tjanez) [reply go/oasis-test-runner/oasis/oasis.go]: Agreed. Done.

File: go/staking/api/sanity_check.go (hunk 6 above)
Comment (tjanez) [reply go/staking/api/sanity_check.go]: Agreed. Done.

**Pull request merged**